ql/src/java/org/apache/hadoop/hive/ql/plan/MoveWork.java

```diff
@@ -26,15 +26,23 @@
 import org.apache.hadoop.hive.ql.exec.Utilities;
 import org.apache.hadoop.hive.ql.hooks.ReadEntity;
 import org.apache.hadoop.hive.ql.hooks.WriteEntity;
+import org.apache.hadoop.hive.ql.io.AcidUtils;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.parse.ImportSemanticAnalyzer;
 import org.apache.hadoop.hive.ql.parse.repl.metric.ReplicationMetricCollector;
+import org.apache.hadoop.hive.ql.plan.BaseCopyWork;
+import org.apache.hadoop.hive.ql.plan.DeferredWorkHelper;
 import org.apache.hadoop.hive.ql.plan.Explain.Level;
 
+import java.util.Collections;
+import java.util.TreeMap;
+
 /**
  * MoveWork.
  *
  */
 @Explain(displayName = "Move Operator", explainLevels = { Level.USER, Level.DEFAULT, Level.EXTENDED })
-public class MoveWork implements Serializable {
+public class MoveWork implements Serializable, BaseCopyWork {
```

Comment:
Doesn't CopyWork and ReplCopyWork also need to implement BaseCopyWork? I don't see it.

Reply:
> Doesn't CopyWork and ReplCopyWork also need to implement BaseCopyWork? I don't see it.

Ah, Thanks for pointing out. I missed it in CopyWork. I will extend CopyWork to implement BaseCopyWork. ReplCopyWork inherits CopyWork. Hence, I think, we only need to override the setupWork in ReplCopyWork.

I will push a new patchset shortly.

Thanks,
Janardhan